sdetools/modules/import_veracode/veracode_integrator.py

```
@@ -35,12 +63,39 @@ def _make_raw_finding(self, node):
                 entry[attr] = node.attributes[attr].value
         return entry
 
+    def parse(self):
+        if self.config['source'] == 'file' or self.config['source'] == 'both':
+            super.parse()
+        else:
+            _raw_findings = []
+            _report_ids = []
+        if self.config['source'] == 'api' or self.config['source'] == 'both':
+            # Mechanism to declare multiple app reports to draw from api
+            report = self.config['api_app']
+            raw_findings, report_id = self.parse_report_file(self.retrieve_report(report), 'raw xml')
+            _raw_findings.extend(raw_findings)
+            if report_id:
+                _report_ids.append(report_id)
+
+        self.findings = _raw_findings
+
+        if _report_ids:
+            self.report_id = ', '.join(_report_ids)
+        else:
+            self.report_id = "Not specified"
+            self.emit.info("Report ID not found in report: Using default.")
+
+
     def parse_report_file(self, report_file, report_type):
-        if report_type != 'xml':
+
+        if report_type != 'xml' and self.config['source'] == 'file':
             raise UsageError("Unsupported file type (%s)" % report_type)
 
         try:
-            base = minidom.parse(report_file)
+            if report_type == 'raw xml':
+                base = report_file
+            else:
+                base = minidom.parse(report_file)
         except Exception:
```

Only catch those exceptions you expect to be thrown. In this case its some XML exception that may get thrown - only catch the bare minimum and let the others propagate